avalara/facade.py

```diff
@@ -73,19 +73,27 @@ def apply_taxes(user, basket, shipping_address, shipping_method, shipping_charge
     shipping_charge.tax = line_taxes['SHIPPING']
 
 
-def submit(order):
+def submit(order, lines=None, line_quantities=None):
     """
     Submit tax information from an order
+    If lines isn't set, all lines in the order will be submitted
+    If line_quantities isn't set, the total quantity for each line will be submitted
     """
+    lines = lines or order.lines.all()
+    line_quantities = line_quantities or [l.quantity for l in lines]
+    for line, qty_to_consume in zip(lines, line_quantities):
+        line.quantity = qty_to_consume
```

Comment:
Partial shipment. This is how the `EventHandler.handle_shipping_event` deals with partial shipment, and we're taking payment on items being shipped -> `def take_payment_for_lines(self, order, lines, line_quantities)`